packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -66,6 +66,18 @@ export interface TableProps {
     writeAutoScaling?: AutoScalingProps;
 }
 
+export interface AttributeProps {
+    /**
+     * The name of an attribute.
+     */
+    name: string;
+
+    /**
+     * The data type of an attribute.
+     */
+    type: KeyAttributeType;
```

Comment:
Does it make sense for this type to still be called `KeyAttributeType`, since you're proposing to use `Attribute` also for things like `nonKeyAttributes`? Should it not be called `AttributeType` instead?

On an unrelated note: does it make sense to default it to `AttributeType.String`?

Reply:
> Does it make sense for this type to still be called `KeyAttributeType`, since you're proposing to use `Attribute` also for things like `nonKeyAttributes`?

You're right. `AttributeType` is an appropriate name here.

> On an unrelated note: does it make sense to default it to `AttributeType.String`?

That makes sense for partition key in most cases, but sort key is a bit different. `AttributeType.Number` is also widely used for query efficiency.